Tests/XMLCoderTests/Box/DataBoxTests.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  DataBoxTests.swift
+//  XMLCoderTests
+//
+//  Created by Vincent Esche on 12/19/18.
+//
+
+import XCTest
+@testable import XMLCoder
+
+class DataBoxTests: XCTestCase {
+    typealias Boxed = DataBox
+    
+    typealias FromXMLString = (String) -> Boxed?
+    
+    let base64: FromXMLString = { xmlString in
+        return Boxed(base64: xmlString)
+    }
+    
+    func testUnbox() {
+        let values: [Boxed.Unboxed] = [
+            Data(base64Encoded: "bG9yZW0gaXBzdW0=")!,
+            Data(),
+        ]
+        
+        for unboxed in values {
+            let box = Boxed(unboxed, format: .base64)
+            XCTAssertEqual(box.unbox(), unboxed)
+        }
+    }
+    
+    func testXMLString() {
+        let values: [(String, FromXMLString)] = [
+            ("bG9yZW0gaXBzdW0=", base64),
+            ("ZG9sb3Igc2l0IGFtZXQ=", base64),
+            ("Y29uc2VjdGV0dXIgYWRpcGlzY2luZyBlbGl0", base64),
+        ]
+        
+        for (string, fromXMLString) in values {
+            let box = fromXMLString(string)!
+            XCTAssertEqual(box.xmlString(), string)
+        }
```

Comment:
sorry, I find this whole part with `FromXMLString` a bit hard to follow. Could this test and others in this test case be simplified like this?

```swift
let testStrings = [ 
  "bG9yZW0gaXBzdW0=",  
  "ZG9sb3Igc2l0IGFtZXQ=",
  "Y29uc2VjdGV0dXIgYWRpcGlzY2luZyBlbGl0",
]

for string in testStrings {
  XCTAssertEqual(Boxed(base64: string).xmlString(), string)
}
```

I understand that it tries to unify tests with tests for other box types, but maybe then it should have a comment clarifying the testing approach and making the typealias visible to all those tests or to make the test much simpler.